### PR TITLE
Issue 1028 - Fixed issue with asset license info loading

### LINKED ISSUE #1028: Adobe Stock asset license information is fetched for the previous page while paging navigation
### Preconditions (*)
Install magneto and adobe stock integration
Have an adobe stock account with licensed images (contact @sivaschenko )

### Steps to reproduce (*)
1. Login to Admin panel
2. Go to Catalog -> Category
3. Select any category
4. Expand Content fieldset
5. Click "Select from Gallery" button next to "Category Image" field to open media gallery
6. Click "Search Adobe Stock" button
7. Log in to Adobe Stock using Sign In link in the top right corner
8. Expand preview for the first image on the page and remember it as `imageId1`
9. Open the browse developer console and look for `XHR` requests on the Networking tab
10. Navigate to the next page
11. Expand preview for the first image on the page and remember it as `imageId2`
12. In the developer console find the last `admin/adobe_stock/license/getlist` request and remember the first id in the `ids` request parameter as `firstRequestId`

### Actual result (*)
`firstRequestId` equals to `imageId1`
Request for the license information is sent with ids from the previous page.

### Expected result (*)
`firstRequestId` should equal to `imageId1`
The license information should be requested for the image ids from the page user is navigating to

### LINKED ISSUE #1106: The "Licensed" label disappears from images in Adobe Stock grid if the Enhanced Media Gallery is enabled
### Preconditions (*)
Magento 2.4-develop
Adobe-Stock-Integration 1.1-develop

1. Installed Magento with Adobe Stock Integration
2. Configured integration in Stores -> Configuration -> Advanced-> System -> Adobe Stock Integration fieldset
3. Enable Enhanced Media Gallery in Stores -> Configuration -> Advanced -> System 
enhansedMG
4. Have licensed images in the Adobe Stock grid

### Steps to reproduce (*)
1. From Admin go to **Content** - **Pages**, click **Add New Page**
2. Expand **Content**,click **Show / Hide Editor**, click **Insert Image...**
3. Click **Search Adobe Stock**
4. Sign In Adobe Stock 
5. Switch between pages and verify the _Licensed_ label

### Expected result (*)
_Licensed_ label on all Licensed images

### Actual result (*)
The _Licensed_ label appears for a while and disappears
licensed_label

### Note
- not reproducible if Enhanced Media Gallery **is disabled**

### PR DESCRIPTION
**Pull request by gabrieldagama**
### Description (*)
The problem was happening when you open AdobeStock gallery and images that were licensed weren't having the label being displayed. 

The problem existed because we were listening to the data provider to fetch licensed information from backend, but when we were building the request we were getting information from the masonry component instead of the data provider. Some times it worked probably because of timing.

Just changed the approach, so now we are listening to the data provider and fetching the items from the data provider.

### Fixed Issues (if relevant)

1. magento/adobe-stock-integration#1028: Adobe Stock asset license information is fetched for the previous page while paging navigation
2. magento/adobe-stock-integration#1106

### Manual testing scenarios (*)
1. Follow the steps on the issue. From a user perspective, we were loading the AdobeStock grid and not having the licensed labels on the licensed images.

### CODE REVIEW

File: AdobeStockImageAdminUi/view/adminhtml/web/js/components/grid/column/overlay.js [hunk 1]
```diff
@@ -79,14 +103,24 @@ define([
         },
 
         /**
-         * Get all ids from data provider
+         * Checks if user is logged in and authorized
+         *
+         * @returns {Boolean}
+         */
+        isUserNotAuthorized: function() {
```
Comment (sivaschenko) [inline AdobeStockImageAdminUi/view/adminhtml/web/js/components/grid/column/overlay.js]: I'd prefer to revert the function to have a better naming
```suggestion
        isUserAuthorized: function() {
```

File: AdobeStockImageAdminUi/view/adminhtml/web/js/components/grid/column/overlay.js [hunk 2]
```diff
@@ -41,19 +43,41 @@ define([
...
+        itemsEventUpdateLicensed: function(items) {
+            var ids = this.getIds(items);
+
+            this.updateLicensed(ids);
+        },
+
+        /**
+         * Updates the licensed data when user logs in.
+         */
+        loginEventUpdateLicensed: function() {
```
Comment (sivaschenko) [inline AdobeStockImageAdminUi/view/adminhtml/web/js/components/grid/column/overlay.js]:
```suggestion
        handleUserUpdate: function() {
```

File: AdobeStockImageAdminUi/view/adminhtml/web/js/components/grid/column/overlay.js [hunk 3]
```diff
@@ -41,19 +43,41 @@ define([
             return this;
         },
 
+        /**
+         * Updates the licensed data when data provider gets updated.
+         *
+         * @param {Array} items
+         */
+        itemsEventUpdateLicensed: function(items) {
```
Comment (sivaschenko) [inline AdobeStockImageAdminUi/view/adminhtml/web/js/components/grid/column/overlay.js]:
```suggestion
        handleItemsUpdate: function(items) {
```

**Review by sivaschenko — CHANGES_REQUESTED** [review]
Thanks for the pull request @gabrieldagama ! Please see my comment regarding the naming

Can you please also check if the change is fixing https://github.com/magento/adobe-stock-integration/issues/1106 and mention it in the description if does

**Review by VladimirZaets — APPROVED** [review]

**Comment by sivaschenko** [comment]
@magento run Functional Tests B2B

**Review by sivaschenko — APPROVED** [review]

**Pull request merged**